nodejs/eks/cluster.ts

```diff
@@ -998,6 +1058,26 @@ export interface ClusterOptions {
      * https://www.pulumi.com/docs/intro/concepts/programming-model/#autonaming
      */
     name?: pulumi.Input<string>;
+
+    /**
+     * The HTTP(S) proxy to use within a proxied environment.
+     *
+     * The proxy is used during cluster creation, and OIDC configuration.
+     *
+     * This is an alternative option to setting the proxy environment variables:
+     * HTTP(S)_PROXY and/or http(s)_proxy.
+     *
+     * This option is required iff the proxy environment variables are not set.
+     *
+     * Format:      <protocol>://<host>:<port>
+     * Auth Format: <protocol>://<username>:<password>@<host>:<port>
+     *
+     * Ex:
+     *   - "http://proxy.example.com:3128"
+     *   - "https://proxy.example.com"
+     *   - "http://username:password@proxy.example.com:3128"
+     */
+    proxy?: string;
```

Comment:
Should this be an Input?

Reply:
It was initially, but the constructor for [https-proxy-agent](https://github.com/TooTallNate/node-https-proxy-agent) can only accept a `string`, or a bag of options for the proxy, with the `string` being the common approach:

https://github.com/TooTallNate/node-https-proxy-agent/blob/3d2981c25a3ff0363a6a38754cfe25b3bf27e5cc/src/index.ts#L8-L12